lib/reunion.rb

```diff
@@ -0,0 +1,78 @@
+require 'pry'
+class Reunion
+  attr_reader :name, :activities
+
+  def initialize(name)
+    @name = name
+    @activities = []
+  end
+
+  def add_activity(activity)
+    @activities << activity
+  end
+
+  def total_cost
+   @activities.sum do |activity|
+      activity.total_cost
+    end
+  end
+
+  def breakout
+   total_owed = Hash.new(0)
+   owned_for_each_activity.each do |hash|
+     hash.map do |key, value|
+       total_owed[key] += value
+     end
+   end
+   total_owed
+  end
+
+  def owned_for_each_activity
+    @activities.map do |activity|
+      activity.owed.inject(Hash.new(0)) do |total_owed_for_activity, (key, value)|
+        total_owed_for_activity[key] = activity.owed
+      end
```

Comment:
Is the idea here to create an array with hashes pointing to their costs? Feel as though there's not much advantage to that over just having a collection of Activities.